File: Components/WinForms/CS/ChartsMainDemo/Modules/RangeViews/RangeArea.cs
using System;
using DevExpress.Utils;
using DevExpress.XtraEditors;

namespace DevExpress.XtraCharts.Demos.RangeViews {
    public partial class RangeAreaDemo : ChartDemoBase2D {

        Series Series { get { return chart.Series.Count > 0 ? chart.Series[0] : null; } }
        RangeAreaSeriesView View { get { return Series != null ? Series.View as RangeAreaSeriesView : null; } }
        
        public override ChartControl ChartControl { get { return this.chart; } }

        public RangeAreaDemo() {
            InitializeComponent();
        }

        void InitializeMarkerKindControls(ComboBoxEdit comboBoxEditMarkerKind, Marker marker) {
            MarkerKindItemCollection itemCollection = MarkerKindItemCollection.CreateCollection();
            MarkerKindItem[] items = new MarkerKindItem[itemCollection.Count];
            itemCollection.CopyTo(items);
            comboBoxEditMarkerKind.Properties.Items.AddRange(items);
            if (marker.Kind != MarkerKind.Star)
                comboBoxEditMarkerKind.SelectedItem = itemCollection.GetItemByKind(marker.Kind);
            else
                comboBoxEditMarkerKind.SelectedItem = itemCollection.GetItemByStarPointCount(marker.StarPointCount);
        }
        void InitializeMarkerSizeControls(ComboBoxEdit comboBoxEditMarkerSize, Marker marker) {
            comboBoxEditMarkerSize.Properties.Items.AddRange(MarkerKindItemCollection.GetSizeArray());
            comboBoxEditMarkerSize.Text = marker.Size.ToString();
        }
        void checkEditShowMinMarkers_CheckedChanged(object sender, EventArgs e) {
            if (View != null)
                View.Marker1Visibility = checkEditShowMinMarkers.Checked ? DefaultBoolean.True : DefaultBoolean.False;
            UpdateControls();
        }
        void checkEditShowMaxMarkers_CheckedChanged(object sender, EventArgs e) {
            if (View != null)
                View.Marker2Visibility = checkEditShowMaxMarkers.Checked ? DefaultBoolean.True : DefaultBoolean.False;
            UpdateControls();
        }
        void comboBoxEditMinMarkerKind_SelectedIndexChanged(object sender, EventArgs e) {
            if (comboBoxEditMinMarkerKind.SelectedIndex == -1)
                return;
            MarkerKindItem item = comboBoxEditMinMarkerKind.SelectedItem as MarkerKindItem;
            if (View != null && item != null) {
                View.Marker1.Kind = item.MarkerKind;
                if (item.MarkerKind.Equals(MarkerKind.Star))
                    View.Marker1.StarPointCount = ((StarMarkerKindItem)item).PointCount;
            }
        }
        void comboBoxEditMaxMarkerKind_SelectedIndexChanged(object sender, EventArgs e) {
            if (comboBoxEditMaxMarkerKind.SelectedIndex == -1)
                return;
            MarkerKindItem item = comboBoxEditMaxMarkerKind.SelectedItem as MarkerKindItem;
            if (View != null && item != null) {
                View.Marker2.Kind = item.MarkerKind;
                if (item.MarkerKind.Equals(MarkerKind.Star))
                    View.Marker2.StarPointCount = ((StarMarkerKindItem)item).PointCount;
            }
        }
        void comboBoxEditMinMarkerSize_SelectedIndexChanged(object sender, EventArgs e) {
            if (comboBoxEditMinMarkerSize.SelectedIndex == -1)
                return;
            if (View != null)
                View.Marker1.Size = Int32.Parse(comboBoxEditMinMarkerSize.Text);
        }
        void comboBoxEditMaxMarkerSize_SelectedIndexChanged(object sender, EventArgs e) {
            if (comboBoxEditMaxMarkerSize.SelectedIndex == -1)
                return;
            if (View != null)
                View.Marker2.Size = Int32.Parse(comboBoxEditMaxMarkerSize.Text);
        }
        void comboBoxLabelKind_SelectedIndexChanged(object sender, EventArgs e) {
            if (Series == null)
                return;
            RangeAreaSeriesLabel label = Series.Label as RangeAreaSeriesLabel;
            if (label == null)
                return;
            switch ((string)comboBoxLabelKind.SelectedItem) {
                case "One Label":
                    label.Kind = RangeAreaLabelKind.OneLabel;
                    break;
                case "Two Labels":
                    label.Kind = RangeAreaLabelKind.TwoLabels;
                    break;
                case "Min Value Label":
                    label.Kind = RangeAreaLabelKind.MinValueLabel;
                    break;
                case "Max Value Label":
                    label.Kind = RangeAreaLabelKind.MaxValueLabel;
                    break;
                case "Value 1 Label":
                    label.Kind = RangeAreaLabelKind.Value1Label;
                    break;
                case "Value 2 Label":
                    label.Kind = RangeAreaLabelKind.Value2Label;
                    break;
                default:
                    label.Kind = RangeAreaLabelKind.TwoLabels;
                    break;
            }
            if (label.Kind == RangeAreaLabelKind.OneLabel)
                label.TextPattern = "{VD:G2}";
            else
                label.TextPattern = "{V:G2}";
        }

        protected override void InitControls() {
            base.InitControls();
            InitializeMarkerSizeControls(comboBoxEditMinMarkerSize, View.Marker1);
            InitializeMarkerSizeControls(comboBoxEditMaxMarkerSize, View.Marker2);
            InitializeMarkerKindControls(comboBoxEditMinMarkerKind, View.Marker1);
            InitializeMarkerKindControls(comboBoxEditMaxMarkerKind, View.Marker2);
            Series.DataSource = SourceOfEnergy.GetEuropeBrentPrices();
        }

        public override void UpdateControls() {
            base.UpdateControls();
            comboBoxLabelKind.Enabled = ShowLabels;
            labelLabelKind.Enabled = ShowLabels;
            comboBoxEditMinMarkerSize.Enabled = checkEditShowMinMarkers.Checked;
            labelMinMarkerSize.Enabled = checkEditShowMinMarkers.Checked;
            comboBoxEditMinMarkerKind.Enabled = checkEditShowMinMarkers.Checked;
            labelMinMarkerKind.Enabled = checkEditShowMinMarkers.Checked;
            comboBoxEditMaxMarkerSize.Enabled = checkEditShowMaxMarkers.Checked;
            labelMaxMarkerSize.Enabled = checkEditShowMaxMarkers.Checked;
            comboBoxEditMaxMarkerKind.Enabled = checkEditShowMaxMarkers.Checked;
            labelMaxMarkerKind.Enabled = checkEditShowMaxMarkers.Checked;
        }
    }
}

File: Components/WinForms/VB/ChartsMainDemo/Modules/RangeViews/RangeArea.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Utils
Imports DevExpress.XtraEditors

Namespace DevExpress.XtraCharts.Demos.RangeViews
	Partial Public Class RangeAreaDemo
		Inherits ChartDemoBase2D

		Private ReadOnly Property Series() As Series
			Get
				Return If(chart.Series.Count > 0, chart.Series(0), Nothing)
			End Get
		End Property
		Private ReadOnly Property View() As RangeAreaSeriesView
			Get
				Return If(Series IsNot Nothing, TryCast(Series.View, RangeAreaSeriesView), Nothing)
			End Get
		End Property

		Public Overrides ReadOnly Property ChartControl() As ChartControl
			Get
				Return Me.chart
			End Get
		End Property

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub InitializeMarkerKindControls(ByVal comboBoxEditMarkerKind As ComboBoxEdit, ByVal marker As Marker)
			Dim itemCollection As MarkerKindItemCollection = MarkerKindItemCollection.CreateCollection()
			Dim items(itemCollection.Count - 1) As MarkerKindItem
			itemCollection.CopyTo(items)
			comboBoxEditMarkerKind.Properties.Items.AddRange(items)
			If marker.Kind <> MarkerKind.Star Then
				comboBoxEditMarkerKind.SelectedItem = itemCollection.GetItemByKind(marker.Kind)
			Else
				comboBoxEditMarkerKind.SelectedItem = itemCollection.GetItemByStarPointCount(marker.StarPointCount)
			End If
		End Sub
		Private Sub InitializeMarkerSizeControls(ByVal comboBoxEditMarkerSize As ComboBoxEdit, ByVal marker As Marker)
			comboBoxEditMarkerSize.Properties.Items.AddRange(MarkerKindItemCollection.GetSizeArray())
			comboBoxEditMarkerSize.Text = marker.Size.ToString()
		End Sub
		Private Sub checkEditShowMinMarkers_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkEditShowMinMarkers.CheckedChanged
			If View IsNot Nothing Then
				View.Marker1Visibility = If(checkEditShowMinMarkers.Checked, DefaultBoolean.True, DefaultBoolean.False)
			End If
			UpdateControls()
		End Sub
		Private Sub checkEditShowMaxMarkers_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkEditShowMaxMarkers.CheckedChanged
			If View IsNot Nothing Then
				View.Marker2Visibility = If(checkEditShowMaxMarkers.Checked, DefaultBoolean.True, DefaultBoolean.False)
			End If
			UpdateControls()
		End Sub
		Private Sub comboBoxEditMinMarkerKind_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEditMinMarkerKind.SelectedIndexChanged
			If comboBoxEditMinMarkerKind.SelectedIndex = -1 Then
				Return
			End If
			Dim item As MarkerKindItem = TryCast(comboBoxEditMinMarkerKind.SelectedItem, MarkerKindItem)
			If View IsNot Nothing AndAlso item IsNot Nothing Then
				View.Marker1.Kind = item.MarkerKind
				If item.MarkerKind.Equals(MarkerKind.Star) Then
					View.Marker1.StarPointCount = (CType(item, StarMarkerKindItem)).PointCount
				End If
			End If
		End Sub
		Private Sub comboBoxEditMaxMarkerKind_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEditMaxMarkerKind.SelectedIndexChanged
			If comboBoxEditMaxMarkerKind.SelectedIndex = -1 Then
				Return
			End If
			Dim item As MarkerKindItem = TryCast(comboBoxEditMaxMarkerKind.SelectedItem, MarkerKindItem)
			If View IsNot Nothing AndAlso item IsNot Nothing Then
				View.Marker2.Kind = item.MarkerKind
				If item.MarkerKind.Equals(MarkerKind.Star) Then
					View.Marker2.StarPointCount = (CType(item, StarMarkerKindItem)).PointCount
				End If
			End If
		End Sub
		Private Sub comboBoxEditMinMarkerSize_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEditMinMarkerSize.SelectedIndexChanged
			If comboBoxEditMinMarkerSize.SelectedIndex = -1 Then
				Return
			End If
			If View IsNot Nothing Then
				View.Marker1.Size = Int32.Parse(comboBoxEditMinMarkerSize.Text)
			End If
		End Sub
		Private Sub comboBoxEditMaxMarkerSize_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEditMaxMarkerSize.SelectedIndexChanged
			If comboBoxEditMaxMarkerSize.SelectedIndex = -1 Then
				Return
			End If
			If View IsNot Nothing Then
				View.Marker2.Size = Int32.Parse(comboBoxEditMaxMarkerSize.Text)
			End If
		End Sub
		Private Sub comboBoxLabelKind_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxLabelKind.SelectedIndexChanged
			If Series Is Nothing Then
				Return
			End If
			Dim label As RangeAreaSeriesLabel = TryCast(Series.Label, RangeAreaSeriesLabel)
			If label Is Nothing Then
				Return
			End If
			Select Case CStr(comboBoxLabelKind.SelectedItem)
				Case "One Label"
					label.Kind = RangeAreaLabelKind.OneLabel
				Case "Two Labels"
					label.Kind = RangeAreaLabelKind.TwoLabels
				Case "Min Value Label"
					label.Kind = RangeAreaLabelKind.MinValueLabel
				Case "Max Value Label"
					label.Kind = RangeAreaLabelKind.MaxValueLabel
				Case "Value 1 Label"
					label.Kind = RangeAreaLabelKind.Value1Label
				Case "Value 2 Label"
					label.Kind = RangeAreaLabelKind.Value2Label
				Case Else
					label.Kind = RangeAreaLabelKind.TwoLabels
			End Select
			If label.Kind = RangeAreaLabelKind.OneLabel Then
				label.TextPattern = "{VD:G2}"
			Else
				label.TextPattern = "{V:G2}"
			End If
		End Sub

		Protected Overrides Sub InitControls()
			MyBase.InitControls()
			InitializeMarkerSizeControls(comboBoxEditMinMarkerSize, View.Marker1)
			InitializeMarkerSizeControls(comboBoxEditMaxMarkerSize, View.Marker2)
			InitializeMarkerKindControls(comboBoxEditMinMarkerKind, View.Marker1)
			InitializeMarkerKindControls(comboBoxEditMaxMarkerKind, View.Marker2)
			Series.DataSource = SourceOfEnergy.GetEuropeBrentPrices()
		End Sub

		Public Overrides Sub UpdateControls()
			MyBase.UpdateControls()
			comboBoxLabelKind.Enabled = ShowLabels
			labelLabelKind.Enabled = ShowLabels
			comboBoxEditMinMarkerSize.Enabled = checkEditShowMinMarkers.Checked
			labelMinMarkerSize.Enabled = checkEditShowMinMarkers.Checked
			comboBoxEditMinMarkerKind.Enabled = checkEditShowMinMarkers.Checked
			labelMinMarkerKind.Enabled = checkEditShowMinMarkers.Checked
			comboBoxEditMaxMarkerSize.Enabled = checkEditShowMaxMarkers.Checked
			labelMaxMarkerSize.Enabled = checkEditShowMaxMarkers.Checked
			comboBoxEditMaxMarkerKind.Enabled = checkEditShowMaxMarkers.Checked
			labelMaxMarkerKind.Enabled = checkEditShowMaxMarkers.Checked
		End Sub
	End Class
End Namespace
